File: 300/300_version1.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int size = nums.length;
        if(size <= 1) return size; // corner case: empty array || one element array
        int[] result = new int[size];
        Arrays.fill(result,1); // fill： initialize the array
        int res = 1;
        for(int curr = 0;curr < size;curr++){
            for(int start = 0;start < curr;start++){
                if(nums[curr] > nums[start]){
                    result[curr] = Math.max(result[start]+1,result[curr]);
                    res = Math.max(res,result[curr]);
                }
            }
        }
        return res;
    }
}

File: 300/300_version1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        size = len(nums)
        if size <=1:return size # corner case, empty list or one element list
        result = [1 for _ in range(size)]
        res = 1
        curr = 0
        while curr < size:
            start = 0
            while start < curr:
                if nums[curr] > nums[start]:
                    result[curr] = max([result[start]+1,result[curr]])
                    res = max([res,result[curr]])
                start+=1
            curr+=1
        return res
